systemverilog-plugin/UhdmAst.cc

```diff
@@ -428,8 +428,26 @@ static void check_memories(AST::AstNode *module_node)
             }
         }
         if (node->type == AST::AST_IDENTIFIER && memories.count(node->str)) {
-            if (!memories[node->str]->attributes.count(UhdmAst::force_convert()) && node->children.size() == 0) {
-                add_force_convert_attribute(memories[node->str]);
+            if (!memories[node->str]->attributes.count(UhdmAst::force_convert())) {
+                bool force_convert = false;
+                // convert memory to list of registers
+                // in case of access to whole memory
+                // or slice of memory
+                // e.g.
+                // logic [3:0] mem [8:0];
+                // always_ff @ (posedge clk) begin
+                //   mem <= '{default:0};
+                //   mem[7:1] <= mem[6:0];
+                // end
+                if (node->children.size() == 0) {
+                    force_convert = true;
+                }
+                if (node->children.size() == 1 && node->children[0]->children.size() != 1) {
+                    force_convert = true;
+                }
```

Comment:
It's unclear to me what these `if`s really do. From what I see, it could either be an identifier with no children, or identifier with one range having both a beginning and an end?
We could have more comments explaining the decision.

Reply:
Yes, you are right, it is identifier with no children (if someone access whole memory), or identifier with one range having both a beginning and an end (if someone access slice of memory).

I've added this to comment and added information what type of memories we don't want to convert.

Reply:
OK, thank you.